processing/src/main/java/org/apache/carbondata/processing/datatypes/ArrayDataType.java

```diff
@@ -171,13 +171,13 @@ public boolean getIsColumnDictionary() {
   }
 
   @Override
-  public void writeByteArray(ArrayObject input, DataOutputStream dataOutputStream,
+  public void writeByteArray(Object input, DataOutputStream dataOutputStream,
       BadRecordLogHolder logHolder, Boolean isWithoutConverter) throws IOException {
-    if (input == null) {
+    if (input == null || input.equals("\\N")) {
```

Comment:
check if we could get `SERIALIZATION_NULL_FORMAT ` while creating complex type from FieldEncoderFactory

Reply:
done